src/middleware/validator/generic.validator.ts

```diff
@@ -12,11 +12,11 @@ export function validate(req: Request, res: Response, next: NextFunction) {
 }
 
 export function idAsInt(req: Request, res: Response, next: NextFunction) {
-  if (!req.params.id) return next()
+  if (!req.params.id) throw new Error('Missing id param on id required request')
```

Comment:
What you have here isn't wrong, it'll do the exact same thing as using a generic response. However the only reason for that is because of how the global error handling middleware is setup. I don't mind relying on that global middleware to handle stuff for us, but a few lines below this we actually just return the 400 if the I'd isn't an int. So throwing an actual error here is a bit of an antipattern with the line 3-4 down from here. 

My suggestion is to just return a generic response and a 400 here rather than relying on the global error handler as it's a bit more explicit. In places where we don't have access to the res object (aka not middleware) I think throwing a new error makes sense. 

I'm open to discussion on this though, cause really I could see it going either way. 

```suggestion
  if (!req.params.id) return res.status(400).json(new Generic response('Missing id param on id required request'))
```

Reply:
Lol. I feel like I'm watching you argue against yourself on this one. This makes sense though. So our protocol will be to return the error immediately when we have access to res and to throw an error when we don't and let the error handler process with res.